File: examples/js/src/rs-standalone.ts
import { ResourceServer } from "./resource-server.js";

// Entrypoint when the resource server runs as a standalone service
// (inside docker compose, alongside Keycloak).
const PORT = Number(process.env.PORT ?? 3000);

async function main(): Promise<void> {
  await ResourceServer.startOn(PORT, {
    "/exec/greet": (args) => {
      const name = typeof args.name === "string" ? args.name : "world";
      return { greeting: `Hello, ${name}!` };
    },
  });
  console.log(`[rs-standalone] ready`);
  // Keep the process alive.
  await new Promise(() => {});
}

main().catch((err) => {
  console.error("[rs-standalone] fatal:", err);
  process.exit(1);
});

File: examples/py/src/aap_demo/rs_standalone.py
"""Entrypoint when the resource server runs as a standalone service
(inside docker compose, alongside Keycloak).
"""

from __future__ import annotations

import os
import sys
from typing import Any, Dict

from .resource_server import ResourceServer


def _greet(args: Dict[str, Any]) -> Dict[str, Any]:
    raw = args.get("name")
    name = raw if isinstance(raw, str) else "world"
    return {"greeting": f"Hello, {name}!"}


def main() -> None:
    port = int(os.environ.get("PORT", "3000"))
    try:
        rs = ResourceServer.start_on(port, {"/exec/greet": _greet})
    except Exception as exc:  # noqa: BLE001
        print(f"[rs-standalone] fatal: {exc}", file=sys.stderr)
        sys.exit(1)
    print("[rs-standalone] ready", flush=True)
    rs.serve_forever()


if __name__ == "__main__":
    main()
